sysutils/pkg/Portfile

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        freebsd pkg 1.10.4
+categories          sysutils archivers
+platforms           darwin
+license             BSD
+maintainers         {@apowers313 ato.ms:apowers} \
+                    openmaintainer
+description         Package management tool for FreeBSD
+long_description    Pkg is the Next Generation package management tool for FreeBSD. Its main goals are to faciliate remote binary package upgrades. It also works with ports without remote binary packages.
+homepage            https://wiki.freebsd.org/pkgng
+
+depends_lib         port:libarchive \
+                    port:openssl
+
+checksums           sha256 ced2627aae44b95a752565f2bc067888a6f4f3c2e856c233fc1567016788f6cb \
+                    rmd160 b30e8e926fa44612347bfcd81f1896bb44e48198 \
+                    size 3398284
+
+test.run            yes
+
+pre-configure {
+    if {![file exists "${worksrcpath}/configure"]} {
+        system -W ${worksrcpath} ./autogen.sh
```

Comment:
This won't really work as-is. Either the file is already there in the tarball you download, in which case this is not needed, or it isn't there, in which case you need to run autoreconf, but it will fail because this port does not declare a dependency on the autoconf ports required to do that.

Additionally, this file only contains `autoreconf -if`; we usually do this by putting
```tcl
use_autoreconf yes
autoreconf.args -fvi
```
into the Portfile, which has the nice side effect of automatically adding the required dependencies.

Reply:
Honestly, I grep'd through the `macports-ports` repo for `autogen.sh` and copied how other packages do it... If the way mentioned here is the preferred way, I can do that.

Reply:
I don't mind the actual invocation using `system`, which is less elegant but certainly works – however, the missing dependencies are going to be a problem, e.g. in trace mode, so I'd prefer if you fixed this.

Reply:
It is fine and good and preferred to run the project's `./autogen.sh` (rather than just running `autoreconf`), since we assume the developer will update that script as needed as their project and build system evolves. But it should be run the way we usually do it, with:

```
use_autoreconf      yes
autoreconf.cmd      ./autogen.sh
autoreconf.args
depends_build-append \
                    port:autoconf \
                    port:automake \
                    port:libtool
```

You must have copied the `if {![file exists "${worksrcpath}/configure"]}` dance from the dovecot2-sieve port; I can find it nowhere else. That port has been uniquely and unusually constructed to allow the maintainer to switch rapidly between manually-packaged stable releases (which include a configure script) and automatically-packaged development versions (which don't). Since the developers of pkg haven't provided any manually-packaged files, not even for stable releases, we will always need to run `autoreconf` via `./autogen.sh`, so there's no point in making its invocation conditional.